File: Week-03/Day-2/Logs.ts
'use strict';

// Read all data from 'log.txt'.
// Each line represents a log message from a web server
// Write a function that returns an array with the unique IP adresses.
// Write a function that returns the GET / POST request ratio.


let fs = require('fs');


function unique(value, index, self) {
  return self.indexOf(value) === index;
};


function uniqueIP(fileName: string) {
  let content: string[] = fs.readFileSync(fileName, 'utf-8').split('\r\n').map(value => value.split('   ')); //Read file, and put every line in an array. Every line in different index. Then split up every line into smaller parts, and put then in one array.In the and we have an array in an array.
  let uniqueIPList = content.map(value => value[1]).filter(unique); //Get the 2 cells in the inner array, and check if it is unique
  console.log(uniqueIPList);
};

function Ratio(fileName: string) {
  let content: string[] = fs.readFileSync(fileName, 'utf-8').split('\r\n').map(value => value.split('   ')); //Read file, and put every line in an array. Every line in different index. Then split up every line into smaller parts, and put then in one array.In the and we have an array in an array.
  let requestArray = content.map(value => value[2]);
  let POSTCount: number = 0;
  requestArray.forEach(function (value) {
    if (value === 'POST /') {
      POSTCount++
    }
  })
  console.log('GET/POST request ratio is : ' + (requestArray.length - POSTCount) / POSTCount);

}
uniqueIP('logs.txt');
Ratio('logs.txt')

File: Java/Week3/Day-1/Logs.java

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;

public class Logs {

  public static void main(String[] args) {
    // Read all data from 'log.txt'.
    // Each line represents a log message from a web server
    // Write a function that returns an array with the unique IP adresses.
    // Write a function that returns the GET / POST request ratio.
    Path filePath = Paths.get("log.txt");
    uniq(filePath);
    ratio(filePath);
  }

  public static void uniq(Path filePath) {
    ArrayList<String> uniq = new ArrayList<>();
    try {
      String[] content = Files.readAllLines(filePath).toString().split(",");
      for (String var : content) {
        if (!uniq.contains(var.substring(26, 39))) {
          uniq.add(var.substring(26, 39));
        }
      }
    } catch (IOException e) {
      System.out.println("Can't find file.");
    }
    System.out.println(uniq);
  }

  public static void ratio(Path filePath) {
    Double ratio = 0.0;
    Double sum = 0.0;
    Double post = 0.0;
    try {
      String[] content = Files.readAllLines(filePath).toString().split(",");
      for (String var : content) {
        if (var.length() == 47) {
          post++;
        }
      }
      sum = Double.valueOf(String.valueOf(content.length));
    } catch (IOException e) {
      System.out.println("Can't find file.");
    }
    ratio = (sum - post) / post;
    System.out.println(ratio);
  }
}